lib/rules/new-module-imports.js

```diff
@@ -40,27 +68,55 @@ module.exports = {
 
           const key = fullName.replace(/^Ember\./, '');
           const match = GLOBALS[key];
-          if (match) {
-            // build replacement import for the warning message
-            let importSpecifier;
-            if (match[1]) {
-              importSpecifier = match[2] ? `{ ${match[1]} as ${match[2]} }` : `{ ${match[1]} }`;
-            } else {
-              importSpecifier = match[2] || key;
-            }
-
-            const replacement = `import ${importSpecifier} from '${match[0]}';`;
-
-            const message = `Use  ${replacement}  instead of using  ${fullName}`;
 
+          if (match) {
+            const message = populateMessage({ fullName, key, match });
             // report the issue to ESLint
             context.report(node, message);
-
             // exit the loop after the first match was found
             break;
           }
         }
       }
     };
+
+    function reportNestedProperties(properties, parent) {
+      properties.forEach((item) => {
+        const match = GLOBALS[`${parent}.${item.key.name}`];
+        const message = populateMessage({
+          key: item.key.name,
+          match,
+          parent,
+          type: item.type
+        });
+
+        context.report(item, message);
+      });
+    }
+
+    function populateMessage(obj) {
+      const isNamespace = EMBER_NAMESPACES.indexOf(`${obj.parent}.${obj.key}`) !== -1;
```

Comment:
this could use a comment. what is this doing and why is it only relevant for the injections?